### DISCUSSION TITLE
Get status of parallel jobs into a single Webhook payload

### DISCUSSION THREAD

**Discussion by benstear** [community/community#26526] [Actions]
<p>I would like to check which of my  matrix runs passed and which didnt, and then use a webhook to send this information to a database.</p>
<p>Right now I have something like this:</p>
<pre><code>name: BIOPYPIR-APPROVED
on: [push,pull_request]
jobs:
  build:
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix:
        python-version: [3.5,3.6,3.7]
        os: [ubuntu-latest, macOS-latest]

        (run workflow...)

    - name: Workflow Webhook Action
      if: startsWith(matrix.os, 'ubuntu')
      uses: distributhor/workflow-webhook@v1
      with:
        webhook_url: ${{ secrets.WEBHOOK_URL }}
        data: |
         { "Job-Status" : "${{job.status}}"
            "Python-Versions" : "${{matrix.python-version}}",
            "OS" : "${{matrix.os}}"
           }
</code></pre>
<p>My goal is to send a single webhook that has the success/fail info for every job in my workflow. What’s the easiest way to do this?</p>
<p>Thanks.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/benstear">@benstear</a> ,</p>
<p>The matrix job status and combination info can be checked from workflow jobs APIs <a href="https://developer.github.com/v3/actions/workflow_jobs/#list-jobs-for-a-workflow-run" rel="nofollow noopener">here</a> with <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener">GITHUB RUN_ID</a>. </p>
<p>You can add a 2nd job which needs matrix job, use ‘if: always()’ to make sure job2 will be executed when matrix job fails, code sample as below, my <a href="https://github.com/weide-zhou/ticket13/runs/576069542?check_suite_focus=true" rel="nofollow noopener">checkrun</a> for your reference.</p>
<pre><code>name: matrixjob
on: [push]

jobs:
  job1:
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-16.04, ubuntu-18.04]
        node: [6, 8]

    steps:
      - if: matrix.os=='ubuntu-16.04'
        name: sleep
        run: |
          sleep 20
      - if: matrix.os=='ubuntu-18.04'
        name: failstep
        run: |
          exit 1
      - uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node }}

  job2:
    needs: [job1]
    if: always()
    runs-on: [ubuntu-latest]
    steps:
      - name: check the jobs # get the matrix job status and combination info
        run: |
          jobstatus=$(curl -X GET -s -u "admin:${{ secrets.PAT3 }}" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq ".jobs[] | {job_status: .conclusion, matrix: .name}")
          echo $jobstatus
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/318dd3e16cda302f55ff69f45ae2ab979ea56906.png" data-download-href="/uploads/short-url/74ng90INl0QVdLoFW9YPxOPjPSe.png?dl=1" title="matrixjobstatus.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181098168-963da5a9-9be1-47eb-81a5-adbc7ff8ed18.png" alt="matrixjobstatus.png" data-base62-sha1="74ng90INl0QVdLoFW9YPxOPjPSe" width="690" height="36" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/318dd3e16cda302f55ff69f45ae2ab979ea56906_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">matrixjobstatus.png</span><span class="informations">1541×82 8.34 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Now you get the info you wanted.</p>
<p>If string $jobstatus contains ‘failure’, you can simply mark the whole matrix job status as ‘fail’ ans tranfer to action ‘distributhor/workflow-webhoo’.</p>
<p>Note: action parameter doesn’t accept string $jobstatus, you can set is as env (<a href="https://github.community/t/how-to-use-system-environment-variable-in-cache-actions-path/17960/2">set-env</a> command)</p>

**Reply by thaysealencar** [reply]
This answer was extremely helpful to me today. 
Thanks so much for sharing it.
I was hitting my head against the wall for so many hours because `needs.job_id.result` would always return `success` even though one of the jobs of my matrix was failing. Calling the GH api and getting the full info about all the jobs in my matrix did the trick. In the api response, the jobs of the matrix had the correct `conclusion` property value.

**Comment by benstear** [comment]
<p>Thank you <a class="mention" href="/weide-zhou">@weide-zhou</a> , this is perfect!</p>
<p>Two questions:</p>
<ol>
<li>
<p>Is it possible to access the step outputs or the artifacts from job 1 in job 2 in the same way?</p>
</li>
<li>
<p>What is the PAT3 secret?</p>
</li>
</ol>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/benstear">@benstear</a> ,</p>
<p>Thanks for your reply!</p>
<ol>
<li>step outputs cannot be shared into different jobs. </li>
</ol>
<p>   The only way to persist data between different jobs is ‘artifacts’. If you use artifacts to get the job status in each matrix parallel job, in job2, you need to download all of the artifacts, get the content and combine them to single file, transfer to action, it looks more complicated than the API.</p>
<ol start="2">
<li>I create a personal access token with admin privilege to the repo, and make it as secrets PAT3.</li>
</ol>
